From c5e5cd7ae484c3f520935c0c3b4292149a856822 Mon Sep 17 00:00:00 2001
From: Marc Mutz <marc.mutz@qt.io>
Date: Tue, 28 Oct 2025 13:27:41 +0100
Subject: [PATCH 544/553] =?UTF-8?q?QString:=20document=20how=20to=20implem?=
 =?UTF-8?q?ent=20the=202024=20German=20=C3=9F=E2=86=92=E1=BA=9E=20rule?=
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

Since 2024, the preferred upper-casing of ß in the German locale is ẞ,
not SS. See https://en.wikipedia.org/wiki/%C3%9F reference [5].

[5] https://www.rechtschreibrat.com/DOX/RfdR_Amtliches-Regelwerk_2024.pdf,
    specifically §25 E3 on p.48 (German only).

Qt currently implements the old semantics, by following
Unicode. Document the work-around needed to get the new behavior.

Task-number: QTBUG-141534
Pick-to: 6.8 6.5
Change-Id: I0fe5c9d4f454f8e4dce84767deecdd9aa55bfab5
Reviewed-by: Paul Wicking <paul.wicking@qt.io>
(cherry picked from commit fa6a010254032b739ce145a1c35b6adcd4f20c60)
Reviewed-by: Qt Cherry-pick Bot <cherrypick_bot@qt-project.org>
---
 src/corelib/text/qstring.cpp | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/src/corelib/text/qstring.cpp b/src/corelib/text/qstring.cpp
index 46c01bf232a..711b70ebf8f 100644
--- a/src/corelib/text/qstring.cpp
+++ b/src/corelib/text/qstring.cpp
@@ -7328,6 +7328,12 @@ QString QString::toCaseFolded_helper(QString &str)
     \note In some cases the uppercase form of a string may be longer than the
     original.
 
+    \note Since 2024, the German language officially prefers to uppercase ß
+    (U+00DF LATIN SMALL LETTER SHARP S) as ẞ (U+1E9E LATIN CAPITAL LETTER SHARP S).
+    Qt's implementation follows Unicode, which still mandates the use of "SS".
+    If you need to implement the new German rules, you need to manually do
+    \c{replace(u'ß', u'ẞ')} \e{before} calling this function.
+
     \sa toLower(), QLocale::toLower()
 */
 
-- 
2.51.2

